api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImpl.java

```diff
@@ -40,6 +50,28 @@
 	@Autowired
 	private ObservationCategoryMap categoryMap;
 	
+	@Override
+	public List<String> getSearchResultUuids(@Nonnull SearchParameterMap theParams) {
+		if (!theParams.getParameters(FhirConstants.LASTN_OBSERVATION_SEARCH_HANDLER).isEmpty()) {
+			
+			Criteria criteria = getSessionFactory().getCurrentSession().createCriteria(typeToken.getRawType());
+			
+			setupSearchParams(criteria, theParams);
+			
+			criteria.setProjection(
+			    Projections.projectionList().add(property("uuid")).add(property("concept")).add(property("obsDatetime")));
+			
+			@SuppressWarnings("unchecked")
+			List<LastnObservationResult> results = (List<LastnObservationResult>) criteria.list().stream()
+			        .map(obs -> new LastnObservationResult((Object[]) obs)).collect(Collectors.toList());
```

Comment:
Not sure if we need either of the castings now. Also, the unchecked warning should go away.

Reply:
We need the `Object[]` casting since criteria list elements are of type `Object` and we want `Object[]`. We need the `List<LastnObservationResult>` since the result after mapping to `LastnObservationResults` is of raw type.

Reply:
We can knock it down to just one cast:

```java
@SuppressWarnings("unchecked")
List<LastnObservationResult> results = ((List<Object[]>) criteria.list()).stream()
		.map(LastnObservationResult::new).collect(Collectors.toList());
```

Reply:
Yes, we can do that. As for the warning, it would still be there because of casting from raw type.